### PR TITLE
Remove frontend plugin from pom.xml when no-client option is selected (#55)

### PR DESCRIPTION
**Pull request by JasonTypesCodes**

### CODE REVIEW

File: generators/server/templates/pom.xml.ejs [hunk 1]
```diff
@@ -44,15 +51,14 @@
         <m2e.apt.activation>jdt_apt</m2e.apt.activation>
         <run.addResources>false</run.addResources>
         <!-- These remain empty unless the corresponding profile is active -->
+<%_ if (databaseType === 'sql') { _%>
```
Comment (willbuck) [inline generators/server/templates/pom.xml.ejs:54]: Is liquibase not used for other DB types? Does `'sql'` include all flavors like H2, MySQL, and Postgres, but exclude like Mongo?

File: generators/server/templates/pom.xml.ejs [hunk 2]
```diff
@@ -31,9 +31,16 @@
...
-        <java.version>1.8</java.version>
-        <node.version>v12.14.0</node.version>
-        <npm.version>6.13.7</npm.version>
+        <java.version><%= JAVA_VERSION %></java.version>
+<%_ if (!skipClient) { _%>
+        <node.version>v<%= NODE_VERSION %></node.version>
+    <%_ if (clientPackageManager === 'npm') { _%>
+        <npm.version><%= NPM_VERSION %></npm.version>
+    <%_ } _%>
+    <%_ if (clientPackageManager === 'yarn') { _%>
```
Comment (willbuck) [inline generators/server/templates/pom.xml.ejs:40]: Have we tested Yarn?

File: generators/server/templates/pom.xml.ejs (hunk 1 above)
Comment (JasonTypesCodes) [reply generators/server/templates/pom.xml.ejs:54]: That's correct

File: generators/server/templates/pom.xml.ejs (hunk 2 above)
Comment (JasonTypesCodes) [reply generators/server/templates/pom.xml.ejs:40]: Not yet, but I saw that the upstream generator supported it, so I pulled it in.  It uses the same Maven plugin as npm

**Pull request merged**